src/builtins.ts

```diff
@@ -658,6 +661,32 @@ export function compileCall(
       if (!type) return module.unreachable();
       return module.i32(type.isManaged ? 1 : 0);
     }
+    case BuiltinSymbols.isVoid: { // isVoid<T>() -> bool
+      let type = evaluateConstantType(compiler, typeArguments, operands, reportNode);
+      compiler.currentType = Type.bool;
+      if (!type) return module.unreachable();
+      return module.i32(type.kind == TypeKind.VOID ? 1 : 0);
+    }
+    case BuiltinSymbols.lengthof: { // lengthof<T>(): i32
+      let type = evaluateConstantType(compiler, typeArguments, operands, reportNode);
+      compiler.currentType = Type.i32;
+      if (type === null) return module.unreachable();
+
+      // Report if there is no call signature
+      let signatureReference = type.signatureReference;
+      if (signatureReference === null) {
+        compiler.error(
+          DiagnosticCode.Type_0_has_no_call_signatures,
+          reportNode.range, "1", (typeArguments ? typeArguments.length : 1).toString(10)
+        );
+        return module.unreachable();
+      }
+
+      let parameterNames = signatureReference.parameterNames;
+      return parameterNames === null
+        ? module.i32(0)
+        : module.i32(parameterNames.length);
```

Comment:
maybe move the condition into `module.i32`

Reply:
Got it. I left the expression on the same line. Is that good for code style?